src/pubsub/ua_pubsub.h

```diff
@@ -86,6 +89,26 @@ typedef struct UA_DataSetWriterSample{
 } UA_DataSetWriterSample;
 #endif
 
+typedef enum {
+    UA_PUBSUB_DATASETMESSAGE_TIMESTAMP,
+    UA_PUBSUB_DATASETMESSAGE_WRITERID,
+    UA_PUBSUB_DATASETMESSAGE_PAYLOAD
+    /* TODO add other defined DSM contents */
+} UA_DataSetMessageField;
+
+typedef struct UA_DataSetMessageOffsetDescription{
+    UA_DataSetMessageField *field;
+    UA_UInt16 fieldSize;
```

Comment:
This should not be the field `size`, but the `offset` starting at the beginning of the NetworkMessage.
Because there are some parts of the message (e.g. bitfield headers) we won't touch and don't see here.
So we cannot compute the start position of the fields only from their length.

Reply:
Will be pushed with the next commit.